### ISSUE TITLE
mount_macfuse: the file system is not available (1)

### ISSUE THREAD

**Issue by haochiche** [osxfuse/osxfuse#791]
Hi 

I'm trying to use sshfs command, but it pop up a window shows:
<img width="328" alt="Screenshot 2021-03-21 at 22 35 37" src="https://user-images.githubusercontent.com/32489318/111920213-7229c680-8a96-11eb-97a2-aa8c6e9006b4.png">
After I click the 'Open system perferences', it give error message:
`mount_macfuse: the file system is not available (1)`

Also the security panel shows nothing. I can't find anything for me to allow..

<img width="780" alt="Screenshot 2021-03-21 at 22 35 55" src="https://user-images.githubusercontent.com/32489318/111920270-b026ea80-8a96-11eb-90de-30db27348c83.png">

I tried to uninstall and re-install, but still the same.

My system version is BigSur 11.2.3. The MacFuse version is 4.1.0.

Many thanks

**Comment by haochiche** [comment]
My issue is the same as this one:
https://github.com/osxfuse/osxfuse/issues/741#issuecomment-752318425

And I followed this method, but still failed to fix it. I can confirm the restricted flag is set, but still, it shows the same problem.

Then I rebooted to the safety mode, this time the error message changed, it now shows 
```
kext load failed: -603946996
mount_macfuse: the file system is not available (255)
```
But again, if I return to the normal mode, the original error message comes back.

**Comment by haochiche** [comment]
then I followed this https://github.com/osxfuse/osxfuse/issues/741#issuecomment-728388877
It seems to work with `csrutil disable`. The Security & Privacy showed up and I clicked allow. 

However, after I enable the csrutil, it shows the same error again and asks me to allow the extension but no any information shows up in the Security & Privacy.
`mount_macfuse: the file system is not available (1)`

**Comment by haochiche** [comment]
It finally working after I set `csrutil enable --without kext`. Not sure why, would be appreciated if anyone has a better solution. 
Many thanks

**Comment by msurdi** [comment]
I had the same problem. What fixed it for me was:

    sudo kextutil /Library/Filesystems/osxfuse.fs/Contents/Extensions/10.16/osxfuse.kext

Then I was asked to reboot, after the reboot it started to work again.

**Comment by BrandoZhang** [comment]
Met the same problem and fixed via updating osxfuse.

**Comment by jrbgit** [comment]
Howdy all -- This problem happened to me while using the current beta release (4.2.0) and downgrading to the current stable release (4.1.2) and rebooting resolved the issue. Hope that helps someone. 

Edit:
Big Sur v. 11.5.2

Edit2:
My issue specifically was:
`kext load failed: -603946996
mount_macfuse: the file system is not available (255)`
and a google search landed me here -- so just dropping this in here in case anyone else finds their way here via a search

**Comment by ilya-kozyr** [comment]
In my case it was vice a versa in comparison to @jrbgit : I had the error
<pre>kext load failed: -603947007
mount_osxfuse: the file system is not available (255)</pre>
when was trying to use `sshfs` 4.1.2 on macOS 11.5.2. After updating to sshfs 4.2.0 the error was gone.

**Comment by MircoT** [comment]
> I had the same problem. What fixed it for me was:
> 
> ```
> sudo kextutil /Library/Filesystems/osxfuse.fs/Contents/Extensions/10.16/osxfuse.kext
> ```
> 
> Then I was asked to reboot, after the reboot it started to work again.

I had the same problem with `macfuse` and this solved my problem too (reboot required):

```bash
sudo kextutil /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
```
Remember to check out the latest extension folder in `Library/Filesystems/macfuse.fs/Contents/Extensions`.

**Comment by nicolettacommins** [comment]
I am still having this issue is well and uninstall/reinstalling macfuse or changing between 4.1.2 and 4.2.0 aren't working. ```csrutil enable --without kext``` worked but I got a warning that this might break in the future. I tried doing this and allowing macfuse, then re-enabling csrutil with kext but the issue came back again. 

I also tried:

```sudo kextutil /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext``` 

as suggested above but then got the following error:

```
Executing: /usr/bin/kmutil load --bundle-path /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
Error Domain=KMErrorDomain Code=27 "Extension with identifiers io.macfuse.filesystems.macfuse not approved to load. Please approve using System Preferences." UserInfo={NSLocalizedDescription=Extension with identifiers io.macfuse.filesystems.macfuse not approved to load. Please approve using System Preferences.}
```

**Comment by changhongyan123** [comment]
I tried 
`sudo kextutil /Library/Filesystems/osxfuse.fs/Contents/Extensions/11/osxfuse.kext`
 and it works for me after rebooting.

**Comment by alexkutsan** [comment]
Have problem after 
```
sudo kextutil /Library/Filesystems/macfuse.fs/Contents/Extensions/10.16/macfuse.kext
Executing: /usr/bin/kmutil load --bundle-path /Library/Filesystems/macfuse.fs/Contents/Extensions/10.16/macfuse.kext
Error Domain=KMErrorDomain Code=27 "Extension with identifiers io.macfuse.filesystems.macfuse not approved to load. Please approve using System Preferences." UserInfo={NSLocalizedDescription=Extension with identifiers io.macfuse.filesystems.macfuse not approved to load. Please approve using System Preferences.}
```
What can I do?

**Comment by jrbgit** [comment]
> Have problem after
> 
> ```
> sudo kextutil /Library/Filesystems/macfuse.fs/Contents/Extensions/10.16/macfuse.kext
> Executing: /usr/bin/kmutil load --bundle-path /Library/Filesystems/macfuse.fs/Contents/Extensions/10.16/macfuse.kext
> Error Domain=KMErrorDomain Code=27 "Extension with identifiers io.macfuse.filesystems.macfuse not approved to load. Please approve using System Preferences." UserInfo={NSLocalizedDescription=Extension with identifiers io.macfuse.filesystems.macfuse not approved to load. Please approve using System Preferences.}
> ```
> 
> What can I do?

I think now you need to go into System Preferences and Allow the application.

**Comment by nicolettacommins** [comment]
I was finally able to fix the problem by uninstalling my institution's security software, then I was able to allow the system extension. I was then able to reinstall the security software and macfuse still works. Hope this helps some people who might be having the same issue.

**Comment by mcat-ee** [comment]
> > I had the same problem. What fixed it for me was:
> > ```
> > sudo kextutil /Library/Filesystems/osxfuse.fs/Contents/Extensions/10.16/osxfuse.kext
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Then I was asked to reboot, after the reboot it started to work again.
> 
> I had the same problem with `macfuse` and this solved my problem too (reboot required):
> 
> ```shell
> sudo kextutil /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
> ```
> 
> Remember to check out the latest extension folder in `Library/Filesystems/macfuse.fs/Contents/Extensions`.

Could you clarify where the repo containing the latest extension dir is? (Unless i'm misundestanding? 

I got this error - I think it could potentially be caused due to my old OSX version ( 10.13.6 )

> $ sudo kextutil /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
> /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext - no compatible dependency found for com.apple.kpi.mach.
> /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext - no compatible dependency found for com.apple.kpi.unsupported.
> /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext - no compatible dependency found for com.apple.kpi.iokit.
> /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext - no compatible dependency found for com.apple.kpi.libkern.
> /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext - no compatible dependency found for com.apple.kpi.bsd.
> /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext - no compatible dependency found for com.apple.kpi.mach.
> /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext - no compatible dependency found for com.apple.kpi.unsupported.
> /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext - no compatible dependency found for com.apple.kpi.iokit.
> /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext - no compatible dependency found for com.apple.kpi.libkern.
> /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext - no compatible dependency found for com.apple.kpi.bsd.
> /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext - no compatible dependency found for com.apple.kpi.mach.
> /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext - no compatible dependency found for com.apple.kpi.unsupported.
> /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext - no compatible dependency found for com.apple.kpi.iokit.
> /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext - no compatible dependency found for com.apple.kpi.libkern.
> /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext - no compatible dependency found for com.apple.kpi.bsd.
> Diagnostics for /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext:
> Dependency Resolution Failures: 
>     Only incompatible kexts found for these libraries: 
>         com.apple.kpi.mach
>         com.apple.kpi.unsupported
>         com.apple.kpi.iokit
>         com.apple.kpi.libkern
>         com.apple.kpi.bsd
> 

And then: 

> $ sudo kextutil /Library/Filesystems/osxfuse.fs/Contents/Extensions/10.16/osxfuse.kext
> Can't stat /Library/Filesystems/osxfuse.fs/Contents/Extensions/10.16/osxfuse.kext - No such file or directory.

**Comment by alex-moreno** [comment]
same issue using Big Sur. Just upgrade and should be fixed

**Comment by rdoubleug** [comment]
New to all of this, however, I can't get this working while using Monterey. `sudo kextutil` command from above just gives me the following error:

`Executing: /usr/bin/kmutil load --bundle-path /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
Error Domain=KMErrorDomain Code=71 "Could not find: Unable to get contents of boot kernel collection collection at /System/Volumes/Preboot/CAF84C27-948A-4311-8A46-3600DB8A51EC/boot/C8C438C2A05729A639B608565C7D47DDEF4748786AC1FAF3FB48BA2EAF593D70FD14FE189BD1F6DC272A96827CC08FBA/System/Library/Caches/com.apple.kernelcaches/kernelcache" UserInfo={NSLocalizedDescription=Could not find: Unable to get contents of boot kernel collection collection at /System/Volumes/Preboot/CAF84C27-948A-4311-8A46-3600DB8A51EC/boot/C8C438C2A05729A639B608565C7D47DDEF4748786AC1FAF3FB48BA2EAF593D70FD14FE189BD1F6DC272A96827CC08FBA/System/Library/Caches/com.apple.kernelcaches/kernelcache}`

The path listed in the error doesn't exist on my machine. There is nothing beyond /boot/ directory that corresponds to the rest of the path.

**Comment by harshitgindra** [comment]
> same issue using Big Sur. Just upgrade and should be fixed

When I run this command `sudo kextutil /Library/Filesystems/macFuse.fs/Contents/Extensions/11/macfuse.kext`, I'm facing the same error as this [post](https://github.com/osxfuse/osxfuse/issues/791#issuecomment-950142994). I'm running ` macFuse 4.2.1` on `MAC OSX Catalina`.

When you say upgrade, do you mean upgrade the `xcode` or something else?
Any other suggestions/thoughts

**Comment by jrbgit** [comment]
> When you say upgrade, do you mean upgrade the `xcode` or something else?
> Any other suggestions/thoughts

I was able to resolve this issue without running any `kextutil` commands by either enabling the beta version or stepping back from the beta version of macFUSE in the System Preferences settings.

To be honest, since then I have moved my dev environment into Visual Studio Code and I no longer use macFUSE. I generally mount large projects and used to use macFUSE with Sublime. It generally worked fine, but I would always notice that sometimes there would be a very long delay when opening large repos. Not to mention the extra steps required to use this setup: Launch terminal, mount the drive, then launch Sublime.

If you are mounting a drive to do dev work, I highly suggest you checkout Visual Studio Code. It handles mounting drives perfectly, it's super fast, and automatically connects to any remote folder that is part of my workspace when I start the app. I never thought I'd switch from Sublime, but honestly VS Code is pretty damn awesome...

**Comment by harshitgindra** [comment]
> a drive to do dev work, I highly suggest you checkout Visual Studi

Thanks for sharing. Unfortunately I'm working on a project where mounting drive is a very important feature and right now, I can't get around without using MacFuse. For development, I completely agree with you on using VSCode all the way. One stop shop.

**Comment by BilyZ98** [comment]
I went to macfuse website and download the macfusion , execute uninstall binary first, and then I execute the install binary 
again.
Now everthing works fine for me .

https://osxfuse.github.io/

**Comment by andyincali** [comment]
I got this to work on an M1 Macbook by booting into Recovery mode (hold down the power button for startup options). In recovery, open Terminal and run this command to add the osxfuse developer ID to the allowed list:
```
spctl kext-consent add 3T5GSNBU6W
```
Once that is done, you can verify it like so:
```
root@mac ~ # spctl kext-consent list
Allowed Team Identifiers:
3T5GSNBU6W
```

After this, I was able to boot into macOS, install osxfuse, and approve the kext when prompted. Note that I did not have to disable SIP:

```
root@mac ~ # csrutil status              
System Integrity Protection status: enabled.
```

Once the kext is whitelisted, you can load it successfully:
```
root@mac~ # kextload /Library/Filesystems/macfuse.fs/Contents/Extensions/12/macfuse.kext          
Executing: /usr/bin/kmutil load -p /Library/Filesystems/macfuse.fs/Contents/Extensions/12/macfuse.kext
```
```
root@mac ~ # kextstat | grep -v com.apple
Executing: /usr/bin/kmutil showloaded
No variant specified, falling back to release
Index Refs Address            Size       Wired      Name (Version) UUID <Linked Against>
  238    0 0xfffffe0006f24000 0x4000     0x4000     io.macfuse.filesystems.macfuse (2097.20) 14248D75-9C85-359E-875E-A704E43AFDEE <7 5 4 3 1> 
```

**Comment by daniilrobnikov** [comment]
Just follow the instructions here: https://github.com/macfuse/macfuse/wiki/Getting-Started
It helped me from start to finish

**Comment by herrlich10** [comment]
My experience might be idiosyncratic, but hope it would also provide some inspiration. I have a similar problem. I'm running a macOS Monterey (12.6.5). I downloaded and installed macFUSE 4.5.0 and then SSHFS 2.5.0 from https://osxfuse.github.io/, and when I tried to mount a remote disk, I got:

kext load failed: -603946986
mount_macfuse: the file system is not available (255)

An LLM consulting suggests that in macOS 12, the error message "kext load failed: -603946986" and "mount_macfuse: the file system is not available (255)" indicates that the kernel extension required for MacFUSE is failing to load. The error code "-603946986" corresponds to kOSKextReturnInvalidArgument in macOS, indicating an invalid argument passed to the kext load command.

The followings are what I have tried without luck:
1. Currently the macFUSE version is the latest. 
2. Uninstall and reinstall macFUSE, with OS reboot many times.
3. Install a different version of macFUSE like 4.4.1 which seems to be the last version before macOS 13 support.
4. I've been using SSHFS before upgrade to Monterey so no "kernel extension was blocked from loading" message was shown and nothing to "Allow" in the "Security & Privacy" preference tab (perhaps I've done that before). 
5. I've also already granted Full Disk Access to the Terminal app. 
6. SIP was on. 
7. Rebooted into the Recovery mode and run `csrutil enable --without kext` but the issue persisted.
8. `sudo kextutil /Library/Filesystems/macfuse.fs/Contents/Extensions/12/macfuse.kext` gave the following error:
```
Executing: /usr/bin/kmutil load --bundle-path /Library/Filesystems/macfuse.fs/Contents/Extensions/12/macfuse.kext
Error Domain=KMErrorDomain Code=31 "Error occurred while building a collection: in /Library/Extensions/spl.kext/Contents/PlugIns/KernelExports.kext/KernelExports: in '/Library/Extensions/spl.kext/Contents/PlugIns/KernelExports.kext/KernelExports' missing __TEXT segment" UserInfo={NSLocalizedDescription=Error occurred while building a collection: in /Library/Extensions/spl.kext/Contents/PlugIns/KernelExports.kext/KernelExports: in '/Library/Extensions/spl.kext/Contents/PlugIns/KernelExports.kext/KernelExports' missing __TEXT segment}
```

A further LLM consulting suggests that 
The error message you provided indicates an issue with loading a kernel extension called "spl.kext" while attempting to load MacFUSE's kernel extension "macfuse.kext." This error message suggests that the "spl.kext" extension is missing the required "__TEXT" segment, causing a failure in the kernel extension collection building process. If you are not actively using the "spl.kext" extension, you can try removing it or disabling it temporarily to see if it resolves the problem with loading MacFUSE. 

Then I move the spl.kext from `/Library/Extensions/spl.kext` to other location, and then run `sudo kextutil /Library/Filesystems/macfuse.fs/Contents/Extensions/12/macfuse.kext`, and it works! I finally am able to mount remote disk with sshfs.

**Comment by bfleischer** [comment]
This issue has bee inactive for a long time, so I'm going to close it. Feel free to open a new issue in case you are still running into problems with loading the kernel extension. See https://github.com/macfuse/macfuse/wiki/Getting-Started for instructions (and troubleshooting tips) on how to enable the macFUSE kernel extension.
